pkg/liqonet/conncheck/conncheck.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conncheck
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+// ConnChecker is a struct that holds the receiver and senders.
+type ConnChecker struct {
+	receiver *Receiver
+	// key is the target cluster ID.
+	senders map[string]*Sender
+	sm      sync.RWMutex
+	conn    *net.UDPConn
+}
+
+// NewConnChecker creates a new ConnChecker.
+func NewConnChecker() (*ConnChecker, error) {
+	addr := &net.UDPAddr{
+		Port: port,
+		IP:   net.ParseIP("0.0.0.0"),
+	}
+	conn, err := net.ListenUDP("udp", addr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to listen on UDP: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to listen on UDP: %w", err)
```

Would also write the address the socket is trying to listen to.